### PR TITLE
Mail integration

### PR DESCRIPTION
**Pull request by B1s9l**
# Pull Requests Checklist

Assignee:

- \[x] code commented?
- \[x\] deactivated things not needed? (e.g. other features which only were prepared)
  - \[x\] alerts that this is todo?
- \[x\] test run localhost?
- \[x\] tested manually all features and edge cases?
- \[x\] documentation added / updated?
- \[x\] pre-commit checks passed?
- \[x\] pipelines passed?

Reviewer:

- \[x\] correct branch?
- \[x\] are there any code smells?
- \[x\] did pipeline pass?
- \[x\] is the code well documented?
- \[ \] tested manually all features and edge cases? *-> don't want to use our available mails for that*

### CODE REVIEW

File: src/components/popups/TeamSetting.tsx [hunk 1]
```diff
@@ -206,6 +209,38 @@ const TeamSettings = ({ isOpen, onClose, onEdit, setIsLeave }) => {
...
+          },
+        }
+      );
+
+      setEmail("");
+      setEmailError(`The invitation has been sent to ${email} !`);
+    } catch (error) {
+      console.error("Failed to send email:", handleError(error));
+      setEmailError("Could not send email");
+    }
```
Comment (5v3nn) [inline src/components/popups/TeamSetting.tsx:241]: handle different kind of input according to api specs

File: src/components/popups/TeamSetting.tsx [hunk 2]
```diff
@@ -206,6 +209,38 @@ const TeamSettings = ({ isOpen, onClose, onEdit, setIsLeave }) => {
     }
   };
 
+  const sendInvitationEmail = async () => {
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (5v3nn) [inline src/components/popups/TeamSetting.tsx:213]: use official owasp regex pattern

**Review by 5v3nn — CHANGES_REQUESTED** [review]

File: src/components/popups/TeamSetting.tsx (hunk 1 above)
Comment (B1s9l) [reply src/components/popups/TeamSetting.tsx]: Should be corrected

File: src/components/popups/TeamSetting.tsx (hunk 2 above)
Comment (B1s9l) [reply src/components/popups/TeamSetting.tsx]: Is corrected

File: src/components/popups/TeamSetting.tsx [hunk 3]
```diff
@@ -206,11 +209,61 @@ const TeamSettings = ({ isOpen, onClose, onEdit, setIsLeave }) => {
...
+        setEmailError(
+          "You are not authorized to send invitations for this team."
+        );
+      } else if (error.response.status === 404) {
+        setEmailError(
+          "Team not found. Please check the team details and try again."
+        );
+      } else if (error.response.status === 503) {
+        setEmailError(
+          "Unable to send email at this time. Mail service is unavailable."
```
Comment (5v3nn) [inline src/components/popups/TeamSetting.tsx:252]: unavailable or email not reachable (e.g. invalid email)

**Review by 5v3nn — APPROVED** [review]
just this small fix, otherwise ok and you can merge
